shell/filebeat-eb-extension.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+# Creates a file at specified location by variable $filebeatextension
+#
+# Required variables:
+# LOGSTASH_CLIENT_KEY: SSL key to authenticate to logstash server
+# LOGSTASH_CLIENT_CERT: SSL certificate in PEM format that matches key
+# LOGSTASH_CA_CERT: SSL certificate of certificate authority logstash is using
+#
+# Optional variables:
+# LOGSTASH_DOC_TYPE: Log type tagging from filebeat to logstash
+# FB_EXTENSION_PATH: Path to filebeat extension configuration to generate
+
+LOGSTASH_DOC_TYPE=${LOGSTASH_DOC_TYPE:-elasticbeanstalk}
+FB_EXTENSION_PATH=${FB_EXTENSION_PATH:-.ebextensions/100-filebeat.config}
+
+indent_cert() {
+    # shellcheck disable=SC2001
+    echo "$1" | sed 's/^/        /'
+}
+
+for var in LOGSTASH_CLIENT_KEY LOGSTASH_CLIENT_CERT LOGSTASH_CA_CERT; do
+    if [ -z "${!var}" ]; then
+       echo "$var is not set, not running filebeat extension"
+       exit 0
+    fi
+done
+
+cat << EOF > "$FB_EXTENSION_PATH"
+files:
+    "/etc/filebeat/filebeat.yml":
+        mode: "0644"
+        owner: root
+        group: root
+        content: |
+          filebeat:
```

Comment:
:yodawg: I heard you like yaml, so I put a yaml file in your bash file with an embedded yaml file in it.